### DISCUSSION TITLE
Make web interface smoother

### DISCUSSION THREAD

**Discussion by Vnzq4Qje3zzU** [simple-login/app#825] [Feature Requests]
Some suggestions:

1. New Custom Alias should be just a simple popup
2. Alias filter/sorting doesn't need to reload the whole page
3. Navigation between tabs (Alias, Subdomains, Mailboxes, etc.) could be done without reloading, like the Cloudflare dashboard
4. Custom domain DNS checks probably don't need to reload the entire page (or at least add a button to check everything at once).
5. Make everything faster - I don't know why, but it's taking more than 5 seconds to render any page (doesn't seem to be a bandwidth problem)

**Comment by nguyenkims** [comment]
Making the loading faster and easier to use is indeed in our roadmap for this year.

The filtering/sorting real-time can be tricky, especially when user has a lot of aliases. In this case we need to fetch the aliases from the backend before displaying them instead of just filtering on the downloaded aliases.

We also focus on the pages that are used the most often which are probably the alias page and the contact page. The subdomain, mailbox, custom domain page are only used during the setup and will be optimized in the second batch.
